src/bartholomew.rs

```diff
@@ -43,6 +44,26 @@ pub fn render(req: Request) -> Result<Response> {
     // Load the site config.
     let mut config: template::SiteInfo = toml::from_slice(&std::fs::read(CONFIG_FILE)?)?;
 
+    let mut dynamic_templates: HashMap<String, (String, String)> = HashMap::new();
+    if let Some(val) = &config.dynamic_templates {
+        for template in val {
+            dynamic_templates.insert(
+                template.0.to_owned(),
+                (template.1.to_owned(), template.2.to_owned()),
+            );
+        }
+    }
+
+    // check if path exists in dynamic templates list
+    let is_dynamic_template: bool = dynamic_templates.contains_key(&path_info as &str);
```

Comment:
I think this whole stanza can be replaced with

```
let dynamic_template_params = dynamic_templates.get(&path_info as &str);
```

I'm not sure what pre-checking the key gains you.